ui/src/pages/Login.tsx

```diff
@@ -32,7 +32,29 @@ interface LoginProps {
 }
 
 export const Login: React.FC<LoginProps> = (props: LoginProps) => {
-  const auth = useAuth();
+  const [loading, setLoading] = React.useState(true);
+  const history = useHistory();
+  React.useLayoutEffect(() => {
+    fetch("/api/check_auth")
+      .then((res) => {
+        if (!res.redirected) {
+          history.replace("/admin");
+        } else {
+          setLoading(false);
+        }
+      })
+      .catch((_) => {
+        setLoading(false);
+      });
+  }, [setLoading, history]);
+
+  if (loading) {
+    return <CircularProgress />;
+  }
+
+  function handleOnClick() {
+    window.location.href = `${window.location.origin}/api/login`;
```

Comment:
for this function, when you have an invalid email and it directs to the error screen, clicking the button again doesn't seem to launch the sign-in flow for me?
![image](https://user-images.githubusercontent.com/19617248/111401120-0fec5280-869f-11eb-9591-a24a392524bc.png)

Reply:
Huh weird, seems to work fine with me.

Reply:
Tried on another browser and doesn't seem to work?

https://user-images.githubusercontent.com/19617248/111402922-73c44a80-86a2-11eb-882d-c6d758a3545e.mov

Reply:
Ah yeah I see, lemme fix